hyperstart/api/grpc/hyperstart.proto

```diff
@@ -30,34 +35,39 @@ service HyperstartService {
 	rpc OnlineCPUMem(OnlineCPUMemRequest) returns (google.protobuf.Empty);
 }
 
-message AddContainerRequest {
+message CreateContainerRequest {
 	Container container = 1;
 	Process init = 2;
+	Spec OCI = 3;
```

Comment:
I agree we need this in order to give all the information related to the container.
However, I am not very familiar with the gRPC code/protocol and I was wondering why we need to specify this Spec in details ? I mean we could use an array of bytes (basically a string) as the third parameter here. From the runtime, we marshal whatever Spec structure we want and from the agent, we unmarshal based on the same structure. I am mentioning this because having the entire spec described here means that we would have to change the protocol if we wanted a new field in the spec.

Reply:
It might be worth pointing out here that this representation implies that the agent process **must** be updated "in lock step" with (at the same time as) the other system gRPC elements. For example, if `oci.proto` is updated to support a new version of the OCI spec (and `Spec.Version` is increased) that will clearly alter the format of the messages meaning the agent won't be able to unpack them unless it too has been updated to handle `OCI.Version`. It won't know that though as, ironically, it won't be able to unpack the `OCI` message to query `Version` and check if it can handle that.

It also potentially implies that the agent might need to support older versions of the spec which don't actually change the representation of `Spec` (but might for example have certain fields that are mandatory whereas a newer version of the spec makes them optional).

The above has an impact on system upgrade: it implies that if the gRPC format changes, all containers need to be stopped (or restarted) to pick up the new gRPC message formats (and associated assets including the new agent).